src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/video-unit-form/video-unit-form.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Moment } from 'moment';
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output } from '@angular/core';
+import { AbstractControl, FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+const getVideoId = require('get-video-id');
+
+export interface VideoUnitFormData {
+    name?: string;
+    description?: string;
+    releaseDate?: Moment;
+    source?: string;
+}
+
+export function extractVideoService(url: string) {
+    if (!url) {
+        return undefined;
+    } else {
+        return getVideoId(url).service;
+    }
+}
```

Comment:
skip the else part, if it's not really needed (see other functions as well):

```suggestion
export function extractVideoService(url: string) {
    if (!url) {
        return undefined;
    }
    return getVideoId(url).service;
}
```

Reply:
done in 93fc0335414f4bba82b740ffbef04bdf8a9745db